luxonis_ml/data/datasets/annotation.py

```diff
@@ -0,0 +1,385 @@
+import json
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import Any, ClassVar, Dict, List, Literal, Optional, Tuple, TypedDict, Union
+
+import numpy as np
+import pycocotools.mask as mask_util
+from PIL import Image, ImageDraw
+from pydantic import Field, model_validator
+from pydantic.types import FilePath, PositiveInt
+from typing_extensions import Annotated, TypeAlias
+
+from luxonis_ml.utils import BaseModelExtraForbid, Registry
+
+from ..utils.enums import LabelType
+
+DATASETS_REGISTRY = Registry(name="datasets")
+
+
+KeypointVisibility: TypeAlias = Literal[0, 1, 2]
+NormalizedFloat: TypeAlias = Annotated[float, Field(ge=0, le=1)]
+"""C{NormalizedFloat} is a float that is restricted to the range [0, 1]."""
+
+ParquetDict = TypedDict(
+    "ParquetDict",
+    {
+        "file": str,
+        "type": str,
+        "created_at": datetime,
+        "class": str,
+        "instance_id": int,
+        "task": str,
+        "annotation": str,
+    },
+)
+
+
+def load_annotation(name: str, js: str, data: Dict[str, Any]) -> "Annotation":
+    return {
+        "ClassificationAnnotation": ClassificationAnnotation,
+        "BBoxAnnotation": BBoxAnnotation,
+        "KeypointAnnotation": KeypointAnnotation,
+        "RLESegmentationAnnotation": RLESegmentationAnnotation,
+        "PolylineSegmentationAnnotation": PolylineSegmentationAnnotation,
+        "ArrayAnnotation": ArrayAnnotation,
+        "LabelAnnotation": LabelAnnotation,
+    }[name](**json.loads(js), **data)
+
+
+class Annotation(ABC, BaseModelExtraForbid):
+    """Base class for an annotation.
+
+    @type task: str
+    @ivar task: The task name. By default it is the string representation of the
+        L{LabelType}.
+    @type class_: str
+    @ivar class_: The class name for the annotation.
+    @type instance_id: int
+    @ivar instance_id: The instance id of the annotation. This determines the order in
+        which individual instances are loaded in L{LuxonisLoader}.
+    @type _label_type: ClassVar[L{LabelType}]
+    @ivar _label_type: The label type of the annotation.
+    """
+
+    _label_type: ClassVar[LabelType]
+
+    task: str
```

Comment:
The label dictionary is now indexed by the `"task"` field instead of by the `LabelType`. This allows to have more annotations of the same type in the labels.